crypto/cms/cms_sd.c

```diff
@@ -341,6 +341,116 @@ static int ossl_cms_add1_signing_cert_v2(CMS_SignerInfo *si,
     return ret;
 }
 
+static const struct {
+    const char *name;
+    int md_nid;
+    int noattr_md_nid;  /* in case of 'without signed attributes' */
+    int noattr_md_a_must; /* noattr_md is a 'MUST' not a 'SHOULD' */
+    int has_msg_update;
+} key2data[] = {
+    { "ML-DSA-44", NID_shake256, NID_sha512,   1, 1, },
+    { "ML-DSA-65", NID_shake256, NID_sha512,   1, 1, },
+    { "ML-DSA-87", NID_shake256, NID_sha512,   1, 1, },
+    { "ED25519",   NID_sha512,   NID_sha512,   1, 0, }, /* RFC 8419 */
+    { "ED448",     NID_shake256, NID_shake256, 1, 0, }, /* RFC 8419 */
+    { "SLH-DSA-SHA2-128f",  NID_sha256,   NID_sha256, 0, 0, }, /* RFC 9814 */
+    { "SLH-DSA-SHA2-128s",  NID_sha256,   NID_sha256, 0, 0, },
+    { "SLH-DSA-SHA2-192f",  NID_sha512,   NID_sha256, 0, 0, },
```

Comment:
`noattr_md_nid` is wrong for the remaiing SLH-DSA algorithms.

RFC 9814 says:

> When
      signed attributes are absent, the digestAlgorithm identifier
      SHOULD match the hash function used in the SLH-DSA tree (as shown
      in the list below).

You were taking SHA-256 from the later line which is saying that SHA-256 is an option when there is a concern for legacy implementations.

> When there is a concern for failures
      with legacy implementations that do not support all hash
      functions, signers MAY choose to always use the SHA-256 algorithm
      identifier as the digestAlgorithm when signed attributes are
      absent.